### PR TITLE
Level Design Functionality for Zenzo

### PR DESCRIPTION
**Pull request by nessa-art**
Delivers level design functionality for Planet Zenzo. Environmental polish can still be improved and added onto for the final deliverable, as well as dialogue and cutscenes.

### CODE REVIEW

File: Cosmic Quest - Order and Chaos/Assets/Scripts/World/Puzzles/LeverPuzzle.cs [hunk 1]
```diff
@@ -1,31 +1,38 @@
...
 
 public class LeverPuzzle : Puzzle
 {
     [Tooltip("List of levers involved with this puzzle")]
     public Lever[] levers;
+    
+    /// <summary>
+    /// List of colours required
+    /// </summary>
+    [SerializeField] protected CharacterColour[] requiredColours;
```
Comment (trudeaua) [inline Cosmic Quest - Order and Chaos/Assets/Scripts/World/Puzzles/LeverPuzzle.cs:13]: This field seems to be unused, mind removing it?

File: Cosmic Quest - Order and Chaos/Assets/Scripts/World/Puzzles/Puzzle.cs [hunk 2]
```diff
@@ -1,34 +1,47 @@
...
+using System.Collections.Generic;
+using UnityEngine;
 using UnityEngine.Events;
 
 public class Puzzle : MonoBehaviour, ISerializable
 {
     public UnityEvent onCompletion;
-    protected CharacterColour[] playerColours;
-
+    public UnityEvent onReset;
```
Comment (trudeaua) [inline Cosmic Quest - Order and Chaos/Assets/Scripts/World/Puzzles/Puzzle.cs:9]: Pretty sure this is the old version of Puzzle, I removed onReset bc I never used it. If you're using it then nvm.

File: Cosmic Quest - Order and Chaos/Assets/Scripts/World/Puzzles/Puzzle.cs [hunk 3]
```diff
@@ -1,34 +1,47 @@
...
+    protected CharacterColour puzzleColour = CharacterColour.None;
```
Comment (trudeaua) [inline Cosmic Quest - Order and Chaos/Assets/Scripts/World/Puzzles/Puzzle.cs:10]: Also unused

File: Cosmic Quest - Order and Chaos/Assets/Scripts/World/Puzzles/Puzzle.cs [hunk 4]
```diff
@@ -1,34 +1,47 @@
...
-    /// </summary>
-    public virtual void ResetPuzzle()
-    {
+    }
+
+    /// <summary>
+    /// Set the colour of the puzzle
+    /// </summary>
+    /// <param name="colour">Colour that the puzzle's elements should be</param>
+    public void SetPuzzleColour(CharacterColour colour)
```
Comment (trudeaua) [inline Cosmic Quest - Order and Chaos/Assets/Scripts/World/Puzzles/Puzzle.cs:35]: This method is also unused. I definitely think this is the old version and maybe git messed up the merge or something

File: Cosmic Quest - Order and Chaos/Assets/Scripts/World/Puzzles/Puzzle.cs (hunk 3 above)
Comment (trudeaua) [inline Cosmic Quest - Order and Chaos/Assets/Scripts/World/Puzzles/Puzzle.cs:10]: Also unused

File: Cosmic Quest - Order and Chaos/Assets/Scripts/World/Puzzles/Puzzle.cs (hunk 2 above)
Comment (nessa-art) [reply Cosmic Quest - Order and Chaos/Assets/Scripts/World/Puzzles/Puzzle.cs:9]: No I'm not using it either. I can remove.

File: Cosmic Quest - Order and Chaos/Assets/Scripts/World/Puzzles/Puzzle.cs (hunk 4 above)
Comment (nessa-art) [reply Cosmic Quest - Order and Chaos/Assets/Scripts/World/Puzzles/Puzzle.cs:35]: Strange...

**Pull request merged**